File: ulohy/37-desitkovy-rozklad-ii/reseni/main.py
if __name__ == "__main__":
    number = input()
    decimal_dot_index = number.find('.')

    if decimal_dot_index == -1:
        decimal_dot_index = len(number)

    for i in range(decimal_dot_index):
        multiple = number[i]

        if multiple == '0':
            continue

        print(f"{multiple} x 1", end='')
        for j in range(decimal_dot_index - i - 1):
            print('0', end='')

        print()

    for i in range(decimal_dot_index + 1, len(number)):
        multiple = number[i]

        if multiple == '0':
            continue

        print(f"{multiple} x 0.", end='')
        for j in range(i - decimal_dot_index - 1):
            print('0', end='')

        print('1')

File: ulohy/37-desitkovy-rozklad-ii/reseni/main.cpp
#include <iostream>
#include <string>

int main() {
    std::string number;
    std::cin >> number;
    int decimalDotIndex = number.find('.');

    if (decimalDotIndex == -1) {
        decimalDotIndex = number.length();
    }

    for (int i = 0; i < decimalDotIndex; i++) {
        char multiple = number[i];

        if (multiple == '0') {
            continue;
        }

        std::cout << multiple << " x 1";
        for (int j = 0; j < decimalDotIndex - i - 1; j++) {
            std::cout << '0';
        }

        std::cout << std::endl;
    }

    for (int i = decimalDotIndex + 1; i < number.length(); i++) {
        char multiple = number[i];

        if (multiple == '0') {
            continue;
        }

        std::cout << multiple << " x 0.";
        for (int j = 0; j < i - decimalDotIndex - 1; j++) {
            std::cout << '0';
        }

        std::cout << '1' << std::endl;
    }

    return 0;
}
